sysvars: add `tidb_shard_row_id_bits` and `tidb_pre_split_regions` (#18917)
pingcap/docs: system-variables.md

- Range: `[1, 100000]`
- This variable controls the maximum number of execution plans that can be cached by [Non-prepared plan cache](/sql-non-prepared-plan-cache.md).

> ### `tidb_pre_split_regions` <span class="version-mark">New in v8.4.0</span>

Reviewer: ```suggestion
### tidb_pre_split_regions <span class="version-mark">New in v8.4.0</span>
```

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Integer
- Default value: `0`
- Range: `[0, 15]`

> - This variable is used to set the default row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically set it when executing CREATE TABLE statements, allowing use of pre_split_regions on such tables (for example NONCLUSTERED tables). See [pre_split_regions](/sql-statements/sql-statement-split-region.md) for details. This variable is typically used in conjunction with `tidb_shard_row_id_bits` to shard new tables and split the regions of new tables.

Reviewer: ```suggestion
- This variable is used to set the default row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically set it when executing `CREATE TABLE` statements, allowing use of `PRE_SPLIT_REGIONS` on such tables (for example NONCLUSTERED tables). For more information, see [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions). This variable is typically used in conjunction with [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-new-in-v840) to shard new tables and split the Regions of new tables.
```

> - This variable is used to set the default row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically set it when executing `CREATE TABLE` statements, allowing use of `PRE_SPLIT_REGIONS` on such tables (for example NONCLUSTERED tables). For more information, see [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions). This variable is typically used in conjunction with [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-new-in-v840) to shard new tables and split the Regions of new tables.

Reviewer 2: ```suggestion
- This variable is used to set the default number of row split shards number for newly created tables. When this variable is set to a non-zero value, TiDB will automatically apply this attribute to tables that allow the use of `PRE_SPLIT_REGIONS` (for example, `NONCLUSTERED` tables) when executing `CREATE TABLE` statements. For more information, see [`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions). This variable is typically used in conjunction with [`tidb_shard_row_id_bits`](/system-variables.md#tidb_shard_row_id_bits-new-in-v840) to shard new tables and pre-split the Regions of new tables.
```

- Range: `[1, 9223372036854775807]`
- This variable controls the maximum number of continuous IDs to be allocated for the [`AUTO_RANDOM`](/auto-random.md) or [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) attribute. Generally, `AUTO_RANDOM` IDs or the `SHARD_ROW_ID_BITS` annotated row IDs are incremental and continuous in one transaction. You can use this variable to solve the hotspot issue in large transaction scenarios.

### `tidb_shard_row_id_bits` <span class="version-mark">New in v8.4.0</span>

- Scope: SESSION | GLOBAL
- Persists to cluster: Yes
- Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): No
- Type: Integer
- Default value: `0`
- Range: `[0, 15]`

> - This variable is used to set the shard bits for new table's default row ID. When this variable has a non-zero value, TiDB will automatically set this property when executing CREATE TABLE statements on tables that allow using `shard_row_id_bits` (such as NONCLUSTERED tables). See [SHARD_ROW_ID_BITS](/shard-row-id-bits.md) for details.

Reviewer: ```suggestion
- This variable is used to set the shard bits for the default row ID of a new table. When this variable has a non-zero value, TiDB will automatically set this property when executing `CREATE TABLE` statements on tables that allow using `SHARD_ROW_ID_BITS` (such as NONCLUSTERED tables). For more information, see [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).
```

> - This variable is used to set the shard bits for the default row ID of a new table. When this variable has a non-zero value, TiDB will automatically set this property when executing `CREATE TABLE` statements on tables that allow using `SHARD_ROW_ID_BITS` (such as NONCLUSTERED tables). For more information, see [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).

Reviewer 2: ```suggestion
- This variable is used to set the default number of row ID shard bits for newly created tables. When this variable is set to a non-zero value, TiDB will automatically apply this attribute to tables that allow the use of `SHARD_ROW_ID_BITS` (for example, `NONCLUSTERED` tables) when executing `CREATE TABLE` statements. For more information, see [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md).
```
